### ISSUE TITLE
Input lag on android while dragging

### ISSUE THREAD

**Issue by kaiec** [godotengine/godot#34480]
**Godot version:** 3.1.2

**OS/device including version:** Arch Linux for dev, Samsung Galaxy S8 for testing, Android 9

**Issue description:** I want to create casual games like board games or where you have to connect dots with lines and so on. When you drag a piece or draw a line using touch, this is usually very immediate, you can drag pieces around as fast as you want and there is no noticeable lag (at least on my phone). 

With Godot, the best I could achieve with a minimal example looks like this: https://imgur.com/a/dMmsktx 

I tried all of the following:
- Using _input with InputEventScreenDrag where the event.position is used: https://github.com/kaiec/punkte/blob/input-only-minimal/scripts/game.gd
- Using _input with InputEventScreenTouch, but this fires only once when the screen is touched, so not usable for dragging
- Pulling in _process and _physics_process: Here I used the mouse emulation, as I could not figure out a way to get the current touch position natively. I tried binding drag or touch events to an action, but this did not work either, I could not use is_action_pressed to determine the touch state and get a position (maybe I am missing something here). 

**Steps to reproduce:** One of the minimal examples is as follows:

```
extends Node2D
var icon = preload("res://assets/icon.png")

func _draw():
		draw_texture(icon, get_local_mouse_position()-Vector2(32,32))	

func _process(delta):
	update()
```
I also used draw_circle, but got the recommendation to try it with a texture, this all did not change the result.

**Minimal reproduction project:** https://github.com/kaiec/punkte/tree/input-only-pull

What I would expect to work best is pulling the current state of the touch screen plus the touch position(s), similar to how the mouse position and state of the buttons can be checked.

**Comment by Calinou** [comment]
I'll state what I said on IRC for reference:

- Latest iOS and Android versions make it possible to disable V-Sync, but this is only advised for drawing applications, not games. I don't know how difficult it would be to implement as a project setting in Godot, but it'd help lower input lag significantly.
- As a stop-gap solution, extrapolating touch input should help decrease perceived latency, at the cost of input being more jittery as a result. This could likely be achieved using a script, and could also be applied to other kinds of input (e.g. mouse input in a FPS).

**Comment by kaiec** [comment]
I just wonder how other Android or iOS games do it, as there is no noticeable lag and that's neither extrapolated (at least I can't imagine that this is the case) and also not a very recent thing. As I also said on IRC, I would think that there is a straight-forward way to get the current touch position. I am glad if I can help, but unfortunately I have no low level Android experience and also only very limited C++ experience.

**Comment by Calinou** [comment]
Does it improve if you set `Input.set_use_accumulated_input(false)` in `_ready()`?

**Comment by dahiyabunty1** [comment]
i have the same problem.
i have tried everything.. but lag on screen drag.while no frame drag other objects. but only screendrage.
while screen drag work smooth on pc

**Comment by kaiec** [comment]
@Calinou: Just checked both with pulling using mouse emulation and with pushing using _input, unfortunately I could not see a difference. Test code in above links is updated.

**Comment by dahiyabunty1** [comment]
hello... i am new to github..i dont know what is happening about issue.. pls tell me when this issue will be solved.... my android game is depend on this... this is only with android

**Comment by Calinou** [comment]
@dahiyabunty1 We don't know when this will be solved, as nobody is currently working on it.

cc @m4gr3d, as he might have knowledge about this.

**Comment by NoFr1ends** [comment]
I’m trying to track down where the delay comes from and what it is exactly.

**Comment by NoFr1ends** [comment]
There is no lag between `gotTouchEvent`, `queueEvent`, `GodotLib.touch` and receiving the event in `_input`. Also I wasn't able to see a delay between received the first event and the last event and when I start the move and stop the move. Currently it looks like the lag or the wrong touch position comes from Android, I'm trying to debug more and look up how android does there pointer tracking.

Also I capture a better video of the issue with some debug logs:
https://i.imgur.com/ZyMXw3k.gifv

**Comment by dahiyabunty1** [comment]
thnks NoFr1ends...
my game is depend on this...
player.global_position = get_global_mouse_position()

**Comment by kaiec** [comment]
According to my further research, this is indeed not Godot specific. My guess is that this is the issue:

https://stackoverflow.com/questions/20981947/what-are-the-causes-of-input-touch-display-lag-in-android

Nevertheless it should be solvable.

**Comment by dahiyabunty1** [comment]
yes.... but i see most android game... player dont lag... they are very quick like stick with finger...so they have solution and much of them are unity games

**Comment by NoFr1ends** [comment]
I'm looking into Android Frame Pacing, but it won't get implemented before 4.0 for sure.

**Comment by dahiyabunty1** [comment]
i beleiive you... i think so... one day godot will turn into best by improving and bug fixing

**Comment by kaiec** [comment]
I am not so sure anymore, that the frame pacing does the trick. I updated both code branches so that according to my understanding, there should be only a redraw every tenth second, which seems to work. However, it still feels like the dot is lagging behind. And with this artificially reduced rendering rate, it should not have anything to do with frame stuffing. Very confusing...

**Comment by 4ian** [comment]
@NoFr1ends Were you able to give a try at implementing Android Frame Pacing library since last December? The team behind the Android Frame Pacing library would be interested to support its integration into Godot, to help solve these input lags on Android.

Let me know if there is a way to help you or anybody that would have been trying this. :)

**Comment by m4gr3d** [comment]
@4ian AFAIK, there hasn't been further development on this issue, so support would be appreciated.
That said, `master` currently doesn't build for Android, so the work may need to be started on the latest stable version (`3.2`) and then ported to `4.0` once it's farther along.

**Comment by NoFr1ends** [comment]
@4ian As @m4gr3d already stated the master isn't buildable on Android currently, also many changes to the Android version is planned because of the deprecation of GLES3 in favor of Vulkan. That's why I didn't continued with testing Android Frame Pacing library.

But after that is done I definitely appreciate help with implementing Android Frame Pacing.

**Comment by 4ian** [comment]
Thanks both for your super fast answers! :)
We'll surely then wait for getting a stable Android build, after the planned changes are done - or give a try at 3.2 later.

@NoFr1ends Feel free to ping me if you take a new look at this later.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

**Comment by Calinou** [comment]
I'll close this as nobody else has been able to reproduce this recently. Please comment (with the [issue template](https://github.com/godotengine/godot/blob/master/.github/ISSUE_TEMPLATE/bug_report.md) filled) if you can still reproduce this.

**Comment by pmellati** [comment]
I'm still experiencing this issue with Godot v4.0.3. Happy to provide video as proof, if necessary.

**Comment by Calinou** [comment]
@pmellati Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

Also, which device model are you using?

**Comment by pmellati** [comment]
Hi @Calinou, here is the info you requested.

# My device
* Model: Samsung galaxy z fold 4
* Android version: 13

# Reproduction & demo

* Minimal project: [drag-input-lap-minimal-demo.zip](https://github.com/godotengine/godot/files/11950282/drag-input-lap-minimal-demo.zip)
* Demo videos
  + With screen capped to 60 Hz:  https://github.com/godotengine/godot/assets/2670251/3eb04b7b-df57-4481-a457-a6e54f4ec3d9
  + With screen uncapped (120 Hz): https://github.com/godotengine/godot/assets/2670251/cc429ae1-0067-4537-96c1-abeaef917536

As you can see in the videos, the icon is falling behind my finger quite a bit.

**Comment by subash-ev** [comment]
is this issue solved (2023) ??

**Comment by subashz** [comment]
@pmellati is this issue been fixed ??

**Comment by Calinou** [comment]
@subashz If you're still running into this issue, please open a new issue with a minimal reproduction project attached.
